c2corg_ui/static/js/documentediting.js

```diff
@@ -0,0 +1,269 @@
+
+  /**
+   * @type {?number}
+   * @private
+   */
+  this.id_ = null;
+
+  /**
+   * @type {?string}
+   * @private
+   */
+  this.culture_ = null;
+
+  if ('appDocumentEditingId' in $attrs &&
+      'appDocumentEditingCulture' in $attrs) {
+    this.id_ = $attrs['appDocumentEditingId'];
+    this.culture_ = $attrs['appDocumentEditingCulture'];
+    // Get document attributes from the API to feed the model:
+    this.http_.get(this.buildUrl_('read')).then(
+        goog.bind(this.successRead_, this),
+        goog.bind(this.errorRead_, this)
+    );
+  }
+};
+
+
+/**
+ * @const
+ * @type {string}
+ */
+app.DocumentEditingController.VIEW_PROJ = 'EPSG:4326';
+
+
+/**
+ * @const
+ * @type {string}
+ */
+app.DocumentEditingController.DATA_PROJ = 'EPSG:3857';
+
+
+/**
+ * @param {string} type Type of URL.
+ * @return {string} URL.
+ * @private
+ */
+app.DocumentEditingController.prototype.buildUrl_ = function(type) {
+  var url = this.apiUrl_;
+  url += url.substr(-1) != '/' ? '/' : '';
+  url += this.module_;
+  switch (type) {
+    case 'read':
+      url += '/' + this.id_ + '?l=' + this.culture_;
+      break;
+    case 'update':
+      url += '/' + this.id_;
+  }
+  return url;
+};
+
+
+/**
+ * @param {Object} response Response from the API server.
+ * @private
+ */
+app.DocumentEditingController.prototype.successRead_ = function(response) {
+  var data = response['data'];
+  if ('geometry' in data && data['geometry']) {
+    var geometry = data['geometry'];
+    if ('geom' in geometry && geometry['geom']) {
+      var geojson = new ol.format.GeoJSON();
+      var point = /** @type {ol.geom.Point} */
+          (geojson.readGeometry(geometry['geom']));
+      point.transform(app.DocumentEditingController.DATA_PROJ,
+                      app.DocumentEditingController.VIEW_PROJ);
+      var coordinates = point.getCoordinates();
+      data['geometry']['geom'] = {
+        'longitude': coordinates[0],
+        'latitude': coordinates[1]
+      };
+    }
+  }
+  this.scope_[this.modelName_] = data;
+};
+
+
+/**
+ * @param {Object} response Response from the API server.
+ * @private
+ */
+app.DocumentEditingController.prototype.errorRead_ = function(response) {
+  // TODO: better error handling
+  alert('Failed loading the document: ' +
+      response.status + ' ' + response.statusText);
+};
+
+
+/**
+ * @param {boolean} isValid True if form is valid.
+ * @export
+ */
+app.DocumentEditingController.prototype.submitForm = function(isValid) {
+  if (!isValid) {
+    // TODO: better handling?
+    alert('Form is not valid');
+  }
+
+  // push to API
+  var data = this.scope_[this.modelName_];
+  if (!goog.isArray(data['locales'])) {
+    // FIXME
+    // With ng-model="route.locales[0].description" route.locales is taken
+    // as an object instead of an array.
+    var locale = data['locales']['0'];
+    delete data['locales'];
+    data['locales'] = [locale];
+  }
+
+  // Convert geometry data
+  if ('geometry' in data && data['geometry']) {
```

Comment:
Actually they are required only for WP. Not for books documents for instance, by the way books won't have any geometry.
And for routes, we might have geometries but they are not mandatory and won't use lon/lat fields.